src/iuse.cpp

```diff
@@ -8068,6 +8068,31 @@ int iuse::hairkit(player *p, item *it, bool, const tripoint&)
     return it->type->charges_to_use();
 }
 
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need soap to use this." ) );
+    } else if ( g->m.has_nearby_water( g->u.pos() ) ) {
+        int pos = g->inv_for_filter( _( "Wash what?" ), []( const item & itm ) {
+            return itm.has_flag( "FILTHY" ); 
+        } );
+    item &mod = p->i_at( pos );
```

Comment:
Misleading indentation here. This would be much more linear (and hence readable) if written as guard clauses:
```c++
if( it->charges < it->type->charges_to_use() ) {
    p->add_msg_if_player( _( "You need soap to use this." ) );
    return 0;
}

if( !g->m.has_nearby_water( p->pos() ) )  {
    p->add_msg_if_player( _( "You need a source of fresh water to use this." ) );
    return 0;
}
```